File: practica2_820574_839304/c++/write.cpp
#include "write.h"
#include <iostream>

Write::Write() {
    nombre = "write";
}

void Write::ejecutar(PilaInts &pila, int &pc) {
    cout << pila.cima() << endl;
    pila.extraer();
    pc++;
}

string Write::nombrar() {
    return nombre;
}

File: practica2_820574_839304/java/Write.java
/*
* Fichero: Write.java
* Autores: Alicia Lázaro Huerta y Manel Jordá Puig Rubio
* Fecha: 27/02/2023
*/
import java.util.Stack;

public class Write extends Instruccion{
    Write(){
        nombre = "write";
    }
    public void ejecutar(Stack<Integer> pila, int[] pc){
        int a;
        a = pila.pop();
        System.out.print(a);
        System.out.println();
        pc[0]++;
    }
}
